go/cobradocs_sync.go

```diff
@@ -22,76 +22,135 @@ import (
-		Head:                github.String(newBranch),
-		Base:                github.String(branch),
-		Body:                github.String(fmt.Sprintf("## Description\nThis is an automated PR to synchronize the cobradocs with %s", pr.GetHTMLURL())),
-		MaintainerCanModify: github.Bool(true),
-	}
-	newPRCreated, _, err := client.PullRequests.Create(ctx, website.Owner, website.Name, newPR)
+		return nil, errors.Wrapf(err, "Failed to run cobradoc sync script in repository %s/%s to %s on Pull Request %d", website.Owner, website.Name, op, prInfo.num)
+	}
+
+	// Create a tree of the commit above using the GitHub API and then commit it.
+	_, commit, err := h.writeAndCommitTree(
+		ctx,
+		client,
+		website,
+		pr,
+		branch,
+		"HEAD",
+		baseTree,
+		parent,
+		fmt.Sprintf("synchronize cobradocs with %s/%s#%d", vitess.Owner, vitess.Name, pr.GetNumber()),
+		op,
+	)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to create Pull Request using branch %s on %s/%s", newBranch, website.Owner, website.Name)
+		return nil, err
 	}
 
-	return newPRCreated, nil
-
+	// Push the branch.
+	if _, _, err := client.Git.UpdateRef(ctx, website.Owner, website.Name, &github.Reference{
+		Ref:    &headRef,
+		Object: &github.GitObject{SHA: commit.SHA},
+	}, true); err != nil {
+		return nil, errors.Wrapf(err, "Failed to force-push %s to %s on Pull Request %s", headBranch, op, pr.GetHTMLURL())
+	}
+
+	switch openPR {
+	case nil:
+		// Create a Pull Request for the new branch.
+		newPR := &github.NewPullRequest{
+			Title:               github.String(fmt.Sprintf("[cobradocs] synchronize with %s (vitess#%d)", pr.GetTitle(), pr.GetNumber())),
+			Head:                github.String(headBranch),
+			Base:                github.String(branch),
+			Body:                github.String(fmt.Sprintf("## Description\nThis is an automated PR to synchronize the cobradocs with %s", pr.GetHTMLURL())),
+			MaintainerCanModify: github.Bool(true),
+		}
+		newPRCreated, _, err := client.PullRequests.Create(ctx, website.Owner, website.Name, newPR)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to create Pull Request using branch %s on %s/%s", headBranch, website.Owner, website.Name)
+		}
+
+		return newPRCreated, nil
+	default:
+		if _, _, err := client.Issues.CreateComment(ctx, website.Owner, website.Name, openPR.GetNumber(), &github.IssueComment{
+			Body: github.String(fmt.Sprintf("PR was force-pushed to resync changes after merge of vitess PR %s. Removing do-not-merge label.", pr.GetHTMLURL())),
+		}); err != nil {
+			return nil, errors.Wrapf(err, "Failed to add PR comment on %s", openPR.GetHTMLURL())
+		}
+
+		// Remove the doNotMerge label.
+		if _, err = client.Issues.RemoveLabelForIssue(ctx, website.Owner, website.Name, openPR.GetNumber(), doNotMergeLabel); err != nil {
+			return nil, errors.Wrapf(err, "Failed to add %s label to %s", doNotMergeLabel, openPR.GetHTMLURL())
+		}
+
+		return openPR, nil
+	}
```

Comment:
We should probably rename the Pull Request to remove the "[DO NOT MERGE]" prefix in the title

Reply:
done :) see 653afbc